include/deal.II/lac/sparse_matrix.h

```diff
@@ -2377,6 +2384,35 @@ SparseMatrix<number>::end (const size_type r)
 }
 
 
+template <typename number>
+template <class StreamType>
+void SparseMatrix<number>::print_matrix_market(StreamType &out,
+					       const number threshold) const
+{
+  Assert(cols != 0, ExcNotInitialized());
+  Assert(val != 0, ExcNotInitialized());
+  Assert( threshold > 0, dealii::ExcMessage("Negative threshold!") ); 
+
+  //Print the header
+  out << "%%MatrixMarket matrix coordinate real general\n";
```

Comment:
I can add complex numbers later.  But for now, do you know of a function that throws the right kind of error I can copy?